### ISSUE TITLE
Documentation about `memory` vs 'job_mem` could be improved

### ISSUE THREAD

**Issue by Andrew-S-Rosen** [dask/dask-jobqueue#634]
The [SLRUMCluster](https://jobqueue.dask.org/en/latest/generated/dask_jobqueue.SLURMCluster.html#dask_jobqueue.SLURMCluster) documentation's description for the `memory` keyword argument is a bit ambiguous when comparing against `job_mem`.

We have:
- `memory`: Total amount of memory per job
- `job_mem`: Amount of memory to request in SLURM. If None, defaults to worker processes * memory

The documentation [states](https://jobqueue.dask.org/en/latest/howitworks.html#workers-vs-jobs) that a "job" is a Slurm allocation. So, this leads us to some potentially confusing docs.

**Comment by guillaumeeb** [comment]
You are right, this is the `memory` documentation that is correct, but not precise enough, this should be something like:

- `memory`: Total amount of memory to be used by all workers inside a job. Used by job queuing system by default as amount of memory per job.
- `job_mem`: Amount of memory to request in SLURM for each job, If None, defaults to memory.

**Comment by maneesh29s** [comment]
I agree with the improvements mentioned here. 
Similarly, can we modify the description of "cores" and "job_cpu" parameters too?

Something like this:

- `cores`: Total number of CPU cores on which all the worker threads inside a job will run. The number of threads per worker process are determined using formula ``cores / processes``. Used by job queuing system by default as amount of CPUs per job.

- `job_cpu`: Number of CPUs to request in SLURM for each job. This option is useful to request more CPUs than total number of worker threads, in cases when a dask task might run multi-threaded using OpenMP or other means. If None, defaults to ``cores``.

**Comment by maneesh29s** [comment]
@guillaumeeb Would like to know your thoughts on the above comment regarding description of CPU parameters.

**Comment by maneesh29s** [comment]
I agree with your suggestion.

And I am happy to raise a PR to close this issue.

**Comment by guillaumeeb** [comment]
Sure, please do so!

**Comment by Andrew-S-Rosen** [comment]
@maneesh29s --- is `cores` the cores _per_ node or the total cores? For instance, if one requests 2 nodes of 48 cores each, should `cores` be 48 or 96? I presume the former.

**Comment by kwanUm** [comment]
Any plan for dask jobqueue to deal with multiple nodes and abstract more of slurm from the user of dask?

**Comment by jacobtomlinson** [comment]
@kwanUm if you want to abstract SLURM from the user then you should take a look at [dask-gateway](https://github.com/dask/dask-gateway/).
